Automattic/woocommerce-payments
# Move BNPL location on product pages to below the product price.

Fixes #8128 

#### Changes proposed in this Pull Request

This PR moves the BNPL text on single product pages to just below the product price. The current behavior is that the BNPL text is below the product price on block-based product pages (correct) but **_below_** the PRBs on legacy product pages (incorrect). By changing the priority on the `woocommerce_single_product_summary` filter to `10`, we can get the BNPL text in the correct location.

One other adjustment I'm making here is to take the `margin-bottom` from the product price container and add it to our BNPL element instead. This will help give a tighter default space between the price and the BNPL text. I'm doing this by getting the computed `margin-bottom`, converting that to a pixel, in case it's specified in `em` or `rem`, then setting that value to a variable used in the `margin-bottom` of our BNPL element. At the same time, I'm using CSS to **only** apply those margin adjustments when our BNPL container follows the price element immediately in the DOM. This prevents weird styling issues if another plugin is inserting something else at priority `10` before we add our BNPL.

A note that on block-based single product pages, this PR should have no effect as the BNPL options are already in the correct place due to how the blocks are handled on this page.

I've tested this on Storefront (legacy product pages) as well as Twenty Twenty-Four (blocks product pages). I've also tested this with custom spacing set on the price element.

Before | After
--- | ---
<img width="400" alt="before" src="https://github.com/Automattic/woocommerce-payments/assets/1558827/7bb7773e-385a-49a8-a04e-3a96bc283605">|<img width="400" alt="after" src="https://github.com/Automattic/woocommerce-payments/assets/1558827/170ec823-9014-48cd-b52f-70fb071a183b">


#### Testing instructions

* Make sure BNPL option(s) are enabled for your merchant store and are visible on the single product page.
* After applying this PR, ensure that the BNPL text has been moved to under the product price.
* Also ensure that the spacing between the product price and BNPL elements, and under the BNPL element looks appropriate (no double margins or wide spaces between the price and BNPL.

-------------------


**Post merge**
